proxy/envoy/watcher.go

```diff
@@ -225,7 +230,9 @@ func generateCertHash(certsDir string) []byte {
 			glog.Warningf("failed to read file %q", filename)
 			continue
 		}
-		h.Write(bs)
+		if err := h.Write(bs); err != nil {
```

Comment:
How bad is it if cert-read or hash-write fails here since that would likely indicate some other major problem in the system? In other words, should generateCertHash return an error in addition to hash?